src/js/program/ad-program-controller.js

```diff
@@ -132,21 +135,17 @@ export default class AdProgramController extends ProgramController {
 
 
     destroy() {
-        const { model } = this;
+        const { model, mediaElement, mediaPool } = this;
 
         model.off();
-        this.mediaPool.recycle();
+        mediaPool.clean(this.mediaElement);
```

Comment:
We should only call `mediaPool.clean` if `Features.backgroundLoading`, since the operation is not necessary when we're only using a single video tag, and when that's the case it would have a noticeable effect.